cmd/noms/diff/print_diff.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package diff
+
+import (
+	"io"
+
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/go/util/writers"
+	"github.com/dustin/go-humanize"
+)
+
+type prefixOp string
+
+const (
+	ADD = "+   "
+	DEL = "-   "
+)
+
+type (
+	printFunc func(w io.Writer, op prefixOp, key, val types.Value) error
+)
+
+// PrintDiff writes a textual reprensentation of the diff from |v1| to |v2|
+// to |w|. If |leftRight| is true then the left-right diff is used for ordered
+// sequences - see Diff vs DiffLeftRight in Set and Map.
+func PrintDiff(w io.Writer, v1, v2 types.Value, leftRight bool) (err error) {
+	// In the case where the diff involves two simple values, just print out the
+	// diff and return. This is needed because the code below assumes that the
+	// values being compared have a parent.
+	if !shouldDescend(v1, v2) {
+		line(w, DEL, nil, v1)
+		return line(w, ADD, nil, v2)
+	}
+
+	dChan := make(chan Difference, 16)
```

Comment:
It seems like a small amount of buffering (at a minimum would be good here so that the processing of the diff can move forward in the event that the printing is stopped or slow for some reason. I wasn't aware that creating a buffer for channels was any more expensive than creating an array. These objects are pretty small.